incubator/hnc/api/v1alpha1/hierarchy_types_conversion.go

```diff
@@ -0,0 +1,49 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	v1a2 "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
+)
+
+// ConvertTo converts from this version v1alpha1 to the Hub version v1alpha2.
+func (src *HierarchyConfiguration) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1a2.HierarchyConfiguration)
+
+	// Spec
+	dst.Spec.AllowCascadingDelete = src.Spec.AllowCascadingDelete
+	dst.Spec.Parent = src.Spec.Parent
+
+	// We don't need to convert status because controllers will update it.
+	dst.Status = v1a2.HierarchyConfigurationStatus{}
+
+	// rote conversion - ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	return nil
+}
+
+// We don't need a conversion from v1alpha2 to v1alpha1. because we will never
+// serve both versions. We serve v1alpha1 in 0.5 and v1alpha2 in 0.6. Upgrading
+// from 0.5 to 0.6 only needs one-way conversion. Downgrading is not supported.
```

Comment:
Ah, I'm sorry, this was a suggestion from Jordan (after the meeting you attended) that adding v1alpha2 would be vastly easier if it were a one-way upgrade. There are a bunch of reasons for this  including:
* Switching from group-version-kind to group-resource in the HNCConfiguration Spec (not a reversible operation)
* Lots of changes to the status fields
* Testing overhead and burden

Jordan's actual recommendation was not to support automatic conversion _at all_ and instead provide instructions to manually upgrade the CRs, or perhaps write a script to do so. We compromised by creating a one-way upgrade process, and we'll add instructions to the release notes about how to backup your CRs in case you need to downgrade later.

@yiqigao217 Can you add something about this to the 0.6 release notes issue?

Reply:
Yeah, I recall the history, just figured we should write it down somewhere :)

Reply:
@yiqigao217 did you mention this in the v1a2 design doc? That plus the 0.6 relnotes should be enough IMO.

Reply:
Yes, I mentioned it in v1a2 design doc and just linked it in the /doc/link.md in this PR. Will comment it under 0.6 release note issue #916 .